server-netty/src/main/java/org/jboss/aerogear/webpush/netty/WebPushFrameListener.java

```diff
@@ -98,10 +98,11 @@ public void onHeadersRead(final ChannelHandlerContext ctx,
                               final int padding,
                               final boolean endStream) throws Http2Exception {
         final String path = headers.path().toString();
-        LOGGER.info("onHeadersRead. streamId={}, method={}, path={}, endstream={}", streamId, headers.method(), path, endStream);
+        final String method = headers.method().toString();
+        LOGGER.info("onHeadersRead. streamId={}, method={}, path={}, endstream={}", streamId, method, path, endStream);
 
-        setPathAndMethod(encoder.connection().stream(streamId), path, headers.method());
-        switch (headers.method().toString()) {
+        streamPaths.put(streamId, path);
```

Comment:
@nmittler could you please clarify, is there a better way to bind a path to the stream and use it in the `onDataRead(...)` method?
Later it works well with `stream.setProperty("webpush.path", path)` and `stream.getProperty("webpush.path")`. But now I can't see how to implement it with `Http2Connection.PropertyKey`, which is generated by `connection().newKey()`.
Thanks!